diff --git a/gcc/config.gcc b/gcc/config.gcc
index 6337fb0..e652c8b 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -1180,7 +1180,7 @@ esac
 # Figure out if we need to enable heap trampolines
 # and variadic functions handling.
 case ${target} in
-aarch64*-*-darwin2*)
+aarch64*-*-darwin*)
   # This applies to arm64 Darwin variadic funtions.
   tm_defines="$tm_defines STACK_USE_CUMULATIVE_ARGS_INIT=1"
   # Executable stack is forbidden.
